### PR TITLE
Add Docstrings to Exported Functions and Variables 

### PR DESCRIPTION
**Pull request by YrrepNoj**
Add (somewhat basic) docstrings to all exportable functions, consts, and package global variables. This is one small step to getting us a cleaner and more standardized codebase.

### CODE REVIEW

File: src/pkg/packager/common.go [hunk 1]
```diff
@@ -94,10 +95,6 @@ func (p *Packager) GetPackageName() string {
 		suffix = "tar"
 	}
 
-	if p.cfg.Pkg.Metadata.Version == "" {
```
Comment (YrrepNoj) [inline src/pkg/packager/common.go:97]: Whoops, this probably shouldn't have been removed.
- [ ] Add it back

File: src/pkg/message/connect.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ func PrintConnectStringTable(connectStrings types.ConnectStrings) {
 
 	if len(connectStrings) > 0 {
 		list := pterm.TableData{{"     Connect Command", "Description"}}
-		// Loop over each connecStrings and convert to pterm.TableData
+		// Loop over each connectStrings and convert to pterm.TableData
```
Comment (Racer159) [inline src/pkg/message/connect.go:20]: dropping each here would make this grammatically correct

File: src/pkg/message/generic.go [hunk 3]
```diff
@@ -6,8 +6,10 @@ package message
 
 import "github.com/pterm/pterm"
 
+// Generic represents an implementation of the io.Writer interface
```
Comment (Racer159) [inline src/pkg/message/generic.go:9]: godot

File: src/pkg/message/generic.go [hunk 4]
```diff
@@ -6,8 +6,10 @@ package message
...
 type Generic struct{}
 
+// Write writes a byte array to stdout and returns the number of bytes written
```
Comment (Racer159) [inline src/pkg/message/generic.go:12]: godot

File: src/pkg/utils/network.go [hunk 5]
```diff
@@ -19,9 +19,11 @@ import (
 	"github.com/defenseunicorns/zarf/src/pkg/message"
 )
 
+// SGETProtocol is the custom protocol used to denote online resources signed with a cosign key.
 const SGETProtocol = "sget://"
 
-func IsUrl(source string) bool {
+// IsUrl returns a boolean indicating if the provided string is a valid URL.
```
Comment (Racer159) [inline src/pkg/utils/network.go:25]: IsURL

File: src/pkg/utils/preflight.go [hunk 6]
```diff
@@ -11,6 +11,7 @@ import (
 	"github.com/defenseunicorns/zarf/src/pkg/message"
 )
 
+// ValidHostname returns a boolean indicating if the provided string is a valid hostname
```
Comment (Racer159) [inline src/pkg/utils/preflight.go:14]: Hostname or HostName (see below)

Comment (Racer159) [reply src/pkg/utils/preflight.go:14]: also godot

**Review by Racer159 — CHANGES_REQUESTED** [review]

**Comment by jeff-mccoy** [comment]
Boo I thought you had already merged this in 😭

**Comment by Racer159** [comment]
Those conflicting files are gross

**Comment by jeff-mccoy** [comment]
yeah this something the intern said they were going to merge into the other PR, but it never made it in......

**Comment by jeff-mccoy** [comment]
So ummm, maybe we close this for now until we have time to look at it again @YrrepNoj ?

**Pull request closed without merging**